modules/javafx.graphics/src/main/java/javafx/scene/SpotLight.java

```diff
@@ -0,0 +1,231 @@
+
+    public final Point3D getDirection() {
+        return direction == null ? DEFAULT_DIRECTION : direction.get();
+    }
+
+    public final ObjectProperty<Point3D> directionProperty() {
+        if (direction == null) {
+            direction = new SimpleObjectProperty<>(this, "direction", DEFAULT_DIRECTION) {
+                @Override
+                protected void invalidated() {
+                    NodeHelper.markDirty(SpotLight.this, DirtyBits.NODE_LIGHT);
+                }
+            };
+        }
+        return direction;
+    }
+
+
+    /**
+     * The angle of the spotlight's inner cone. Surfaces whose angle to the light's origin is less than this angle
+     * receive the full light's intensity. At larger angles, the light intensity starts to drop.
+     * <p>
+     * The angle is clamped between 0 and 180.
+     *
+     * @defaultValue 0
+     */
+    private DoubleProperty innerAngle;
+
+    public final void setInnerAngle(double value) {
+        innerAngleProperty().set(value);
+    }
+
+    private static final double DEFAULT_INNER_ANGLE = NGSpotLight.getDefaultInnerAngle();
+
+    public final double getInnerAngle() {
+        return innerAngle == null ? DEFAULT_INNER_ANGLE : innerAngle.get();
+    }
+
+    public final DoubleProperty innerAngleProperty() {
+        if (innerAngle == null) {
+            innerAngle = getLightDoubleProperty("innerAngle", DEFAULT_INNER_ANGLE);
+        }
+        return innerAngle;
+    }
+
+
+    /**
+     * The angle of the spotlight's outer cone. Surfaces whose angle to the light's origin is greater than this angle
+     * receive no light. At smaller angles, the light intensity starts to increase.
+     * <p>
+     * The angle is clamped between 0 and 180.
+     *
+     * @defaultValue 90
+     */
+    private DoubleProperty outerAngle;
+
+    public final void setOuterAngle(double value) {
+        outerAngleProperty().set(value);
+    }
+
+    private static final double DEFAULT_OUTER_ANGLE = NGSpotLight.getDefaultOuterAngle();
+
+    public final double getOuterAngle() {
+        return outerAngle == null ? DEFAULT_OUTER_ANGLE : outerAngle.get();
+    }
+
+    public final DoubleProperty outerAngleProperty() {
+        if (outerAngle == null) {
+            outerAngle = getLightDoubleProperty("outerAngle", DEFAULT_OUTER_ANGLE);
+        }
+        return outerAngle;
+    }
+
+
+    /**
+     * The intensity falloff factor of the spotlight's outer cone. Surfaces whose angle to the light's origin is greater
+     * than the inner angle but less than the outer angle receive partial intensity governed by this factor. The larger
+     * the falloff, the sharper the drop in intensity from the inner cone. A falloff factor of 1 gives a linear drop in
+     * intensity, values greater than 1 give a convex drop, and values smaller than 1 give a concave drop. Negative
+     * values are allowed, but give unrealistic lighting.
+     *
+     * @defaultValue 1
```

Comment:
Maybe it should default to 0? I don't have a strong opinion on this either.

I do think that it should say that values < 0 are clamped to 0.

Reply:
I think that having a dropoff as a default is closer to what users will want. 0 will just give a flat circle (thought there is decreased intensity with the angle to the surface).

Is there a good reason to limit it? Nothing changes if we allow <0 values. I can remove the documentation for it, but in terms of implementation I don't see the benefit.

Reply:
I think a default of 1 is fine then. Also, I don't have the same concern over negative falloff values as I do for the inner and outer angles.